File: C++/C++Basic/04.Conditional-Statements-Exercise/04.toyShop.cpp
#include <iostream>

using namespace std;

int main(){
    double priceExcursion;
    int countPuzzles, countDolls, countBears, countMinions, countTrucks;
    cin >> priceExcursion;
    cin >> countPuzzles >> countDolls >> countBears >> countMinions >> countTrucks;

    double sumOfToys = (countPuzzles * 2.60) 
    + (countDolls * 3) 
    + (countBears * 4.10)
    + (countMinions * 8.20)
    + (countTrucks * 2.0);

    int totalCountToys = countPuzzles + countDolls + countBears + countMinions + countTrucks;

    if (totalCountToys >= 50) {
        sumOfToys *= 0.75;
    }

    sumOfToys *= 0.90;
    
    double moneyLeft = sumOfToys - priceExcursion;
    
    cout.setf(ios::fixed);
    cout.precision(2);

    if (sumOfToys >= priceExcursion){
        cout << "Yes! " << moneyLeft << " lv left." << endl;
    }
    else{
        double neededMoney = priceExcursion - sumOfToys;

        cout << "Not enough money! " << neededMoney << " lv needed." << endl;
    }
   
    return 0;


}

File: Python/Basic/04.Conditional-Statements-Exercise/04.toy_shop.py
price_tour = float(input())
count_puzzle = int(input())
count_dolls = int(input())
count_bears = int(input())
count_minions = int(input())
count_trucks = int(input())

total_sum = (count_puzzle * 2.60)\
            + (count_dolls * 3)\
            + (count_bears * 4.10) \
            + (count_minions * 8.20) \
            + (count_trucks * 2)
total_count_toys = count_puzzle + count_dolls + count_minions + count_bears + count_trucks

if total_count_toys >= 50:
    total_sum *= 0.75
total_sum *= 0.9

if total_sum >= price_tour:
    print(f'Yes! {abs(total_sum-price_tour):.2f} lv left.')
else:
    print(f'Not enough money! {abs(total_sum-price_tour):.2f} lv needed.')